src/vs/workbench/contrib/stats/electron-browser/workspaceStats.ts

```diff
@@ -164,16 +164,18 @@ export class WorkspaceStats implements IWorkbenchContribution {
 		this.reportProxyStats();
 
 		const diagnosticsChannel = this.sharedProcessService.getChannel('diagnostics');
-		diagnosticsChannel.call('reportWorkspaceStats', this.getWorkspaceInformation());
+		this.getWorkspaceInformation().then(stats => diagnosticsChannel.call('reportWorkspaceStats', stats));
 	}
 
-	private getWorkspaceInformation(): IWorkspaceInformation {
+	private async getWorkspaceInformation(): Promise<IWorkspaceInformation> {
 		const workspace = this.contextService.getWorkspace();
 		const state = this.contextService.getWorkbenchState();
-		const id = this.workspaceStatsService.getTelemetryWorkspaceId(workspace, state);
+		const telemetryId = this.workspaceStatsService.getTelemetryWorkspaceId(workspace, state);
+		const rendererSessionId = (await this.telemetryService.getTelemetryInfo()).sessionId;
```

Comment:
mixing await and thens in the same file. let's either update the whole file or stick with then

Reply:
there are still other then calls in this function and the whole file. i think its best to avoid the await altogether